src/dotnet/commands/dotnet-tool/restore/ToolRestoreCommand.cs

```diff
@@ -96,81 +98,110 @@ public override int Execute()
                 return 0;
             }
 
-            Dictionary<RestoredCommandIdentifier, RestoredCommand> dictionary =
```

Comment:
Please review with split diff. The github is confused so bad. I made all "output" collection to be concurrent version. And run an extract method to make it obvious that all the input and output are either readonly or concurrent collection